web/packages/core/src/public/v1.ts

```diff
@@ -0,0 +1,173 @@
+
+    /**
+     * The effective config loaded with the last call to `load()`.
+     * If no such call has been made, this will be `null`.
+     */
+    readonly loadedConfig: URLLoadOptions | DataLoadOptions | null;
+
+    /**
+     * Indicates the readiness of the playing movie.
+     *
+     * @returns The `ReadyState` of the player.
+     */
+    get readyState(): ReadyState;
+
+    /**
+     * The metadata of the playing movie (such as movie width and height).
+     * These are inherent properties stored in the SWF file and are not affected by runtime changes.
+     * For example, `metadata.width` is the width of the SWF file, and not the width of the Ruffle player.
+     *
+     * @returns The metadata of the movie, or `null` if the movie metadata has not yet loaded.
+     */
+    get metadata(): MovieMetadata | null;
+
+    /**
+     * Reloads the player, as if you called {@link RufflePlayer.load} with the same config as the last time it was called.
+     *
+     * If this player has never been loaded, this method will return an error.
+     */
+    reload(): Promise<void>;
+
+    /**
+     * Loads a specified movie into this player.
+     *
+     * This will replace any existing movie that may be playing.
+     *
+     * @param options One of the following:
+     * - A URL, passed as a string, which will load a URL with default options.
+     * - A [[URLLoadOptions]] object, to load a URL with options.
+     * - A [[DataLoadOptions]] object, to load data with options.
+     * The options, if provided, must only contain values provided for this specific movie.
+     * They must not contain any default values, since those would overwrite other configuration
+     * settings with a lower priority (e.g. the general RufflePlayer config).
+     *
+     * The options will be defaulted by the [[config]] field, which itself
+     * is defaulted by a global `window.RufflePlayer.config`.
+     */
+    load(options: string | URLLoadOptions | DataLoadOptions): Promise<void>;
+
+    /**
+     * Plays or resumes the movie.
+     */
+    play(): void;
+
+    /**
+     * Whether this player is currently playing.
+     *
+     * @returns True if this player is playing, false if it's paused or hasn't started yet.
+     */
+    get isPlaying(): boolean;
+
+    /**
+     * Returns the master volume of the player.
+     *
+     * The volume is linear and not adapted for logarithmic hearing.
+     *
+     * @returns The volume. 1.0 is 100% volume.
+     */
+    get volume(): number;
+
+    /**
+     * Sets the master volume of the player.
+     *
+     * The volume should be linear and not adapted for logarithmic hearing.
+     *
+     * @param value The volume. 1.0 is 100% volume.
+     */
+    set volume(value: number);
+
+    /**
+     * Checks if this player is allowed to be fullscreen by the browser.
+     *
+     * @returns True if you may call [[enterFullscreen]].
+     */
+    get fullscreenEnabled(): boolean;
```

Comment:
Isn't this name misleading? I think this could easily be confused to with "isFullscreen".

Reply:
Doesn't it match https://developer.mozilla.org/en-US/docs/Web/API/Document/fullscreenEnabled?

Reply:
tbh I don't entirely agree with having this method. It's a document thing, not at all related to this player or element. The source is `return document.fullscreenEnabled || document.webkitFullscreenEnabled` and whilst I guess webkit prefix is certainly a consideration... is it our place? It just feels weird to have a method like that.